src/interactive/app/eventloop.rs

```diff
@@ -40,18 +43,30 @@ pub enum ProcessingResult {
 }
 
 impl AppState {
+    pub fn navigation_mut(&mut self) -> &mut NavigationState {
+        self.glob_mode.as_mut().unwrap_or(&mut self.normal_mode)
```

Comment:
Enum would make sense if the states were exclusive, however in this case I want to keep the navigation state of the "normal" mode so that when user quits the glob mode he is back to the view where he started.

I am open for suggestions if this can be improved.

Reply:
Ah, I see, thanks for elaborating!
I will see what I can do during the review, which should be the time to actually understand it all. Most of the code right now I only glossed over, and the codebase in general is by now pretty foreign to me.

In any case, it's a very exciting feature and I'd like to land it before or *on* the 24th if possible :D.

Reply:
My plan is to start digging on from the 23rd, and wrap it up (pun absolutely intended) on the 24th :). To me it's fine to fix the wonky cursor some other time as well as it doesn't prevent the feature from working.